prog/rand.go

```diff
@@ -737,6 +744,9 @@ func (r *randGen) generateArg(s *state, typ sys.Type, dir ArgDir, sizes map[stri
 		return dataArg([]byte(a.Val)), constArg(uintptr(len(a.Val))), nil
 	case sys.IntType:
 		v := r.randInt()
+		if a.IsRange == true {
```

Comment:
There is already IntKind that controls generation.
Please add IntRange kind instead of IsRange bool, and then handle it in the switch below.